js/client/modules/@arangodb/testutils/result-processing.js

```diff
@@ -350,18 +348,11 @@ function saveToJunitXML(options, results) {
         state.xml.elem('skipped/', true);
         state.xml.elem('/testcase');
       } else if (!state.seenTestCases) {
-        if (testSuite.failed === 0) {
-          state.xml.elem('testcase', {
-            name: 'all_tests_in_' + state.xmlName,
-            time: 0 + testSuite.duration
-          }, true);
-        } else {
-          state.xml.elem('testcase', {
-            name: 'all_tests_in_' + state.xmlName,
-            failures: testSuite.failuresFound,
-            time: 0 + testSuite.duration
-          }, true);
-        }
+        const elem = addOptionalDuration({ name: 'all_tests_in_' + state.xmlName }, testSuite);
+        if (testSuite.failed !== 0 || testSuite.failed !== undefined) {
```

Comment:
Condition looks strange, should it rather be
```suggestion
        if (testSuite.failed !== 0 && testSuite.failed !== undefined) {
```
Reason: if `testSuite.failed === undefined` it is also `!== 0`.